x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas/schemas.ts

```diff
@@ -24,6 +24,14 @@ export const language = Joi.string().valid('kuery', 'lucene');
 export const output_index = Joi.string();
 export const saved_id = Joi.string();
 export const timeline_id = Joi.string();
+export const timeline_title = Joi.alternatives().when(Joi.ref('$timeline_id.eType'), {
+  is: Joi.string()
+    .not()
+    .empty(),
+  then: Joi.string()
+    .allow('')
+    .required(),
+});
```

Comment:
Since the validation rule is:

timeline_id and timeline_title are both optional but they have to exist together or not exist at all and timeline_title can be an empty string, then this is the Joi rule that would be the prescribed recipe:

```ts
export const timeline_title = Joi.string()
  .allow('')
  .when('timeline_id', {
    is: Joi.exist(),
    then: Joi.required(),
    otherwise: Joi.forbidden(),
  });
``` 

Once you have that you should update the test of:
https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas/create_rules_schema.test.ts#L1067

to be truthy since that test will fail which is what we want. I added comments to the other tests which should remove the `null` statement. Those tests should begin passing as well.